includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,512 @@
+	public function get_themes() {
+		if ( null !== $this->themes ) {
+			return $this->themes;
+		}
+
+		$themes   = $this->get_default_supported_reader_themes();
+		$themes   = array_map( [ $this, 'prepare_theme' ], $themes );
+		$themes[] = $this->get_classic_mode();
+
+		/**
+		 * Filters supported reader themes.
+		 *
+		 * @param array $themes [
+		 *     Reader theme data.
+		 *     {
+		 *         @type string         $name           Theme name.
+		 *         @type string         $slug           Theme slug.
+		 *         @type string         $slug           URL of theme preview.
+		 *         @type string         $screenshot_url The URL of a mobile screenshot. Note: if this is empty, the theme may not display.
+		 *         @type string         $homepage       A link to a page with more information about the theme.
+		 *         @type string         $description    A description of the theme.
+		 *         @type string|boolean $requires       Minimum version of WordPress required by the theme. False if all versions are supported.
+		 *         @type string|boolean $requires_php   Minimum version of PHP required by the theme. False if all versions are supported.
+		 *         @type string         $download_link  A link to the theme's zip file. If empty, the plugin will attempt to download the theme from wordpress.org.
+		 *     }
+		 * ]
+		 */
+		$themes = array_filter( (array) apply_filters( 'amp_reader_themes', $themes ) );
+
+		foreach ( $themes as &$theme ) {
+			$theme['availability'] = $this->get_theme_availability( $theme );
+		}
+
+		$this->themes = $themes;
+
+		return $this->themes;
+	}
+
+	/**
+	 * Gets a reader theme by slug.
+	 *
+	 * @param string $slug Theme slug.
+	 * @return array Theme data.
+	 */
+	public function get_reader_theme( $slug ) {
+		return current(
+			array_filter(
+				$this->get_themes(),
+				static function ( $theme ) use ( $slug ) {
+					return $theme['slug'] === $slug;
+				}
+			)
+		);
+	}
+
+	/**
+	 * Retrieves theme data.
+	 *
+	 * @param boolean $from_api Whether to return theme data from the wordpress.org API. Default false.
+	 * @return array Theme data from the wordpress.org API.
+	 */
+	public function get_default_supported_reader_themes( $from_api = false ) {
+		if ( ! $from_api ) {
+			$themes = self::DEFAULT_READER_THEMES;
+
+			return $themes;
+		}
+
+		// Note: This can be used to refresh the hardcoded raw theme data.
+		if ( ! function_exists( 'themes_api' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/theme.php';
+		}
+
+		$response = themes_api(
+			'query_themes',
+			[
+				'author'   => 'wordpressdotorg',
+				'per_page' => 24, // There are only 12 as of 05/2020.
+			]
+		);
+
+		if ( ! $response || is_wp_error( $response ) ) {
+			return [];
+		}
+
+		if ( is_array( $response ) ) {
+			$response = (object) $response;
+		}
+
+		$supported_themes = array_diff(
+			AMP_Core_Theme_Sanitizer::get_supported_themes(),
+			[ 'twentyten' ] // Excluded because not responsive.
+		);
```

Comment:
Might as well replace this with:

```suggestion
		$supported_themes = wp_list_pluck( self::DEFAULT_READER_THEMES, 'slug' );
```